src/scripts/main.js

```
@@ -1,5 +1,39 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const spider = document.querySelector('.spider');
```

You should check if the `spider` element exists before using it. If `document.querySelector('.spider')` returns null, the code will throw an error when accessing `spider.getBoundingClientRect()`. The requirements specify that the code should fail gracefully if elements are missing.